### PR TITLE
fix: fix non-null constraint

### PR DESCRIPTION
**Pull request by JackTan25**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR
1. for snowflake and postgres, it will return a non-null error when fill default value for non-null field.
2. in this pr, when we try to fill default expr for a non-null field, if its default expr is none, we will return the error
- Closes #issue
https://github.com/datafuselabs/databend/issues/13932

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13939)

### CODE REVIEW

File: src/query/service/src/pipelines/processors/transforms/transform_resort_addon_without_source_schema.rs [hunk 1]
```diff
@@ -48,6 +50,25 @@ pub fn build_expression_transform(
...
+            // but for pg or snowflake, if a field is non-null, it will return
+            // a non-null error (it means they will give a null as the default value).
+            if !f.is_nullable()
+                && f.default_expr().is_none()
+                && !matches!(f.data_type(), DataType::Bitmap)
+            {
+                // if we have a user-specified default expr, it must satisfy the non-null constraint
+                // in table-create phase. So we just consider default_expr is none.
+                return Err(ErrorCode::BadArguments(format!(
+                    "unable to cast type `null` value to a non-null type field `{}`",
```
Comment (sundy-li) [inline src/query/service/src/pipelines/processors/transforms/transform_resort_addon_without_source_schema.rs:68]: The error msg is wrong, 
```
null value in column "{}" violates not-null constraint
```

**Comment by sundy-li** [comment]
This change is correct, but it is breaking changes. It may cause errors for old scripts or sink processes (insert by other drivers).
cc @BohuTANG

**Comment by JackTan25** [comment]
when run script `tests/suites/0_stateless/14_clickhouse_http_handler/14_0007_http_clickhouse_input_format_diagnostic.sh`,we will
get a stack error, so the reason is below:
```rust
split_tx.send(Ok(Split {
    info: split_info,
    rx: data_rx,
}))
.await
.expect("fail to send split from stream"); // do expect here
```
let me remove the bad test added by my.

**Review by b41sh — APPROVED** [review]

**Review by TCeason — APPROVED** [review]

**Comment by JackTan25** [comment]
cc @BohuTANG merge into wizard test is passed, can we merge this?

**Comment by BohuTANG** [comment]
A12/A13 need be fixed to make them works for Databend and Snowflake, I will do it this weekend. @JackTan25

**Comment by BohuTANG** [comment]
> A12/A13 need be fixed to make them works for Databend and Snowflake, I will do it this weekend. @JackTan25

Fixex: https://github.com/datafuselabs/wizard/commit/9cd4d96e9f2a075918e252fa68123531bb54ae53

**Pull request merged**